vars/mavenExecuteStaticCodeChecks.groovy

```diff
@@ -0,0 +1,48 @@
+import com.sap.piper.PiperGoUtils
+import com.sap.piper.Utils
+import groovy.transform.Field
+
+import static com.sap.piper.Prerequisites.checkScript
+
+@Field String METADATA_FILE = 'metadata/mavenStaticCodeChecks.yaml'
+@Field String STEP_NAME = getClass().getName()
+@Field String METADATA_FOLDER = '.pipeline' // metadata file contains already the "metadata" folder level, hence we end up in a folder ".pipeline/metadata"
+
+
+void call(Map parameters = [:]) {
+    handlePipelineStepErrors(stepName: STEP_NAME, stepParameters: parameters) {
+
+        final script = checkScript(this, parameters) ?: null
+
+        if (!script) {
+            error "Reference to surrounding pipeline script not provided (script: this)."
+        }
+        def utils = parameters.juStabUtils ?: new Utils()
+        new PiperGoUtils(this, utils).unstashPiperBin()
+
+        // The parameters map in provided from outside. That map might be used elsewhere in the pipeline
+        // hence we should not modify it here. So we create a new map based on the parameters map.
```

Comment:
I would modify this comment a bit:
```
// Make a shallow copy of the passed-in Map in order to prevent removal of top-level keys
// to be visible in calling code, just in case the map is still used there.
```

Reply:
fyi github introduced multiline suggestions, so you could have done your comment as a suggestion and thus make it simpler to apply